In [2]:
import pandas as pd

In [3]:
detection_counts = pd.read_csv("data/most_detected_tags.csv")
dwell_loc = pd.read_csv("data/loc_code_detection_patterns.csv")
dwell_antenna = pd.read_csv("data/antenna_detection_patterns.csv")
location = pd.read_csv("data/location_lat_long.csv")

In [4]:
detection_counts

tag_id   count in_cleaning_table
0     989.001042  273253                no
1     989.001039  111089                no
2     989.001042   89789                no
3     989.001040   82246                no
4     989.001039   78683                no
...          ...     ...               ...
6781  989.001045       1                no
6782  989.001041       1                no
6783  989.001042       1                no
6784  989.001041       1                no
6785  989.001045       1                no

[6786 rows x 3 columns]

In [5]:
dwell_loc

tag_id loc_code  start_date    end_date  \
0       989.001039       5A  2012-01-01  2023-05-27   
1       900.226001       83  2012-01-01  2012-01-01   
2       989.002011       84  2012-01-01  2012-01-01   
3       982.000412       83  2012-01-01  2012-01-01   
4       989.001006       84  2012-01-01  2012-01-01   
...            ...      ...         ...         ...   
227809  989.001034      401  2025-02-01  2025-02-01   
227810  989.001034      402  2025-02-05  2025-02-05   
227811  989.001045      202  2025-02-05  2025-02-05   
227812  989.001045      201  2025-02-05  2025-02-05   
227813  989.002028      702  2025-02-06  2025-02-06   

                        start_time                    end_time  \
0       2012-01-01 00:00:00.560000  2023-05-27 13:21:01.150000   
1       2012-01-01 00:00:26.290000  2012-01-01 00:22:17.810000   
2       2012-01-01 00:00:47.390000  2012-01-01 00:06:32.050000   
3       2012-01-01 00:00:53.740000  2012-01-01 00:00:53.740000   
4       2012-01-01 00:00:54.890000  2012-01-01 00:01:15.650000   
...                            ...                         ...   
227809  2025-02-01 01:12:03.700000  2025-02-01 01:12:03.700000   
227810  2025-02-05 01:07:23.460000  2025-02-05 01:07:23.460000   
227811  2025-02-05 01:11:04.690000  2025-02-05 01:11:04.690000   
227812  2025-02-05 01:14:39.260000  2025-02-05 01:14:39.260000   
227813  2025-02-06 00:19:28.940000  2025-02-06 02:39:46.360000   

                       dwell_time  number_of_detections  
0       4164 days 13:21:00.590000                 18623  
1          0 days 00:21:51.520000                     8  
2          0 days 00:05:44.660000                     4  
3                 0 days 00:00:00                     1  
4          0 days 00:00:20.760000                     2  
...                           ...                   ...  
227809            0 days 00:00:00                     1  
227810            0 days 00:00:00                     1  
227811            0 days 00:00:00                     1  
227812            0 days 00:00:00                     1  
227813     0 days 02:20:17.420000                     3  

[227814 rows x 8 columns]

# Summaries

## For an individual tag

In [6]:
tag = 989.001006676104

print("Tag ID is:", tag)
print("Total number of detections:", detection_counts[detection_counts["tag_id"] == tag]["count"].iloc[0])
print("Found in cleaning table? ", detection_counts[detection_counts["tag_id"] == tag]["in_cleaning_table"].iloc[0])
print("First detected:", min(dwell_loc[dwell_loc.tag_id == tag]["start_date"]))
print("Last detected:", max(dwell_loc[dwell_loc.tag_id == tag]["end_date"]))

print("Details of each detection:")
tag_dwell = dwell_loc[dwell_loc.tag_id == tag]
tag_dwell

Tag ID is: 989.001006676104
Total number of detections: 78120
Found in cleaning table?  no
First detected: 2022-05-08
Last detected: 2023-05-16
Details of each detection:


tag_id loc_code  start_date    end_date  \
32494  989.001007      402  2022-05-08  2022-06-28   
58612  989.001007      401  2022-06-28  2022-06-28   
58616  989.001007      402  2022-06-28  2022-06-28   
58619  989.001007      401  2022-06-29  2022-06-29   
58628  989.001007      402  2022-06-29  2022-06-29   
...           ...      ...         ...         ...   
72692  989.001007      402  2023-05-14  2023-05-15   
73414  989.001007      401  2023-05-15  2023-05-15   
74383  989.001007      402  2023-05-16  2023-05-16   
74384  989.001007      401  2023-05-16  2023-05-16   
74389  989.001007      402  2023-05-16  2023-05-16   

                       start_time                    end_time  \
32494  2022-05-08 03:48:48.770000  2022-06-28 18:29:55.220000   
58612  2022-06-28 18:40:58.040000  2022-06-28 19:06:18.400000   
58616  2022-06-28 21:14:57.270000  2022-06-28 21:14:57.270000   
58619  2022-06-29 00:13:00.200000  2022-06-29 05:26:20.220000   
58628  2022-06-29 05:30:39.580000  2022-06-29 10:43:41.870000   
...                           ...                         ...   
72692  2023-05-14 21:31:49.480000  2023-05-15 20:28:24.390000   
73414  2023-05-15 20:34:21.440000  2023-05-15 21:36:19.510000   
74383  2023-05-16 04:39:44.090000  2023-05-16 04:39:44.090000   
74384  2023-05-16 04:44:27.310000  2023-05-16 04:44:27.310000   
74389  2023-05-16 05:02:57.270000  2023-05-16 05:18:59.830000   

                    dwell_time  number_of_detections  
32494  51 days 14:41:06.450000                    14  
58612   0 days 00:25:20.360000                     2  
58616          0 days 00:00:00                     1  
58619   0 days 05:13:20.020000                     2  
58628   0 days 05:13:02.290000                    10  
...                        ...                   ...  
72692   0 days 22:56:34.910000                     2  
73414   0 days 01:01:58.070000                     2  
74383          0 days 00:00:00                     1  
74384          0 days 00:00:00                     1  
74389   0 days 00:16:02.560000                     2  

[343 rows x 8 columns]

# What does number of detections mean? What counts as one detection?

SELECT tag_id, datetime, detection_date, detection_time, loc_code <br>
FROM detections <br>
WHERE  tag_id = '989.001006676104' and loc_code = '402' <br>
ORDER BY datetime <br>

If the detections are close to reach other, one detection is every 5 minutes. Sometimes, we don't see them for a bit and then they are detected on the same antenna a few hours later.